20_classes_lab/03.py

```diff
@@ -1,13 +1,66 @@
-"""
-Following code assumes a class named Widget that depends on
-other widgets
-Widget ctor takes a name, and function 
-add_dependency(depends)
-marks widget as depends on all widgets in "depends" list
-
-When building a widget, you need to first build all the widgets it
-depends on.
-
-Implement widget so the following code works
-"""
-
+"""
+Fill the code to print the result
+"""
+
+class Widget(object):
+
+    widget_list = []

```

Comment:
Solution works but it's not recommended. Keeping a list of all the created widgets will keep them in memory even after nobody needs them. If you insist on using such a list it's better to use weak ref:
https://docs.python.org/2/library/weakref.html

However even with weakref it's not such a great idea - You still need to iterate the list and search for each widget to see if it was built.

A better pattern is to use normal instance variables here. 

Please modify the code not to use class variables